gson/src/test/java/com/baeldung/gson/jsoncompare/JsonCompareUnitTest.java

```diff
@@ -0,0 +1,93 @@
+    }
+
+    @Test
+    public void givenIdenticalArrays_whenCompared_thenEqual() {
+        JsonParser parser = new JsonParser();
+        String string1 = "[10, 20, 30]";
+        String string2 = "[10, 20, 30]";
+
+        Assert.assertTrue(parser.parse(string1)
+            .isJsonArray());
+        Assert.assertEquals(parser.parse(string1), parser.parse(string2));
+    }
+
+    @Test
+    public void givenArraysInDifferentOrder_whenCompared_thenNotEqual() {
+        JsonParser parser = new JsonParser();
+        String string1 = "[20, 10, 30]";
+        String string2 = "[10, 20, 30]";
+
+        Assert.assertTrue(parser.parse(string1)
+            .isJsonArray());
+        Assert.assertNotEquals(parser.parse(string1), parser.parse(string2));
+    }
+
+    @Test
+    public void givenIdenticalNestedObjects_whenCompared_thenEqual() {
+        JsonParser parser = new JsonParser();
+        String string1 = "{\"customer\": {\"id\": \"44521\",\"fullName\": \"Emily Jenkins\", \"age\": 27, \"consumption_info\" : {\"fav_product\": \"Coke\", \"last_buy\": \"2012-04-23\"}}}";
+        String string2 = "{\"customer\": {\"id\": \"44521\",\"fullName\": \"Emily Jenkins\", \"age\": 27, \"consumption_info\" : {\"last_buy\": \"2012-04-23\", \"fav_product\": \"Coke\"}}}";
+
+        JsonElement json1 = parser.parse(string1);
+        JsonElement json2 = parser.parse(string2);
+
+        Assert.assertTrue(json1.isJsonObject());
+        Assert.assertEquals(json1, json2);
+    }
+
+    @Test
+    public void givenIdenticalNestedObjectsWithArray_whenCompared_thenEqual() {
+        JsonParser parser = new JsonParser();
+        String string1 = "{\"customer\": {\"id\": \"44521\",\"fullName\": \"Emily Jenkins\", \"age\": 27, \"consumption_info\" : {\"last_buy\": \"2012-04-23\", \"prouducts\": [\"banana\", \"eggs\"]}}}";
+        String string2 = "{\"customer\": {\"id\": \"44521\",\"fullName\": \"Emily Jenkins\", \"age\": 27, \"consumption_info\" : {\"last_buy\": \"2012-04-23\", \"prouducts\": [\"banana\", \"eggs\"]}}}";
+
+        JsonElement json1 = parser.parse(string1);
+        JsonElement json2 = parser.parse(string2);
+
+        Assert.assertTrue(json1.isJsonObject());
+        Assert.assertEquals(json1, json2);
+    }
+
+    @Test
+    public void givenNestedObjectsDifferentArrayOrder_whenCompared_thenNotEqual() {
+        JsonParser parser = new JsonParser();
+        String string1 = "{\"customer\": {\"id\": \"44521\",\"fullName\": \"Emily Jenkins\", \"age\": 27, \"consumption_info\" : {\"last_buy\": \"2012-04-23\", \"prouducts\": [\"banana\", \"eggs\"]}}}";
+        String string2 = "{\"customer\": {\"id\": \"44521\",\"fullName\": \"Emily Jenkins\", \"age\": 27, \"consumption_info\" : {\"last_buy\": \"2012-04-23\", \"prouducts\": [\"eggs\", \"banana\"]}}}";
+
+        JsonElement json1 = parser.parse(string1);
+        JsonElement json2 = parser.parse(string2);
+
+        Assert.assertTrue(json1.isJsonObject());
```

Comment:
I don't think we need to repeatedly assert `isJsonObject` in all the tests. After the first one, we've proved the point. Let's only have it for the first time it's demonstrated.